sources/web/datalab/polymer/components/item-list/item-list.ts

```diff
@@ -251,12 +270,83 @@ class ItemListElement extends Polymer.Element {
       const shadow = '0px ' + yOffset + 'px 10px -5px #ccc';
       headerContainer.style.boxShadow = shadow;
     });
+
+    // Initial sort
+    this._sortBy(0);
   }
 
   resetFilter() {
     this._filterString = '';
   }
 
+  _formatColumnValue(value: ColumnType, i: number, columns: Column[]): string {
+    if (columns[i]) {
+      if (columns[i].type === ColumnTypeName.DATE) {
+        return (value as Date).toLocaleString();
+      } else {
+        return value.toString();
+      }
+    } else {
+      return '';
+    }
+  }
+
+  _columnButtonClicked(e: any) {
+    this._sortBy(e.model.itemsIndex);
+  }
+
+  _sortBy(column: number) {
+    // If sort is requested on the current sort column, reverse the sort order.
+    // Otherwise, set the current sort column to that.
+    if (this._currentSort.column === column) {
+      this._currentSort.asc = !this._currentSort.asc;
+    } else {
+      this._currentSort = {
+        asc: true,
+        column,
+      };
+    }
+
+    this.$.list.sort = (a: ItemListRow, b: ItemListRow) => {
+      // Bail out of sort if no columns have been set yet.
+      if (!this.columns.length) {
+        return;
```

Comment:
Doesn't typescript complain about have a no-value return in a function that returns a value?

Reply:
Unless you define a return type, the compiler will compute the return type for this function to be `number|void`, which is reasonable to me since at this point there's missing information, and the function shouldn't return a number.

Reply:
What behavior do we want or expect from the caller in this case? It seems unlikely the caller is expecting to get back a void value, so it could end of doing something strange. Perhaps this function should return 0 in this case (we define all rows as being equal, meaning we don't care how it gets sorted), or throw an exception.

Reply:
I missed this reply. Sure this sounds reasonable. I can include it in the next change to item-list, which can be your other suggestion of optionally using relative time for Date columns.